python/lsst/pipe/base/pipeline.py

```diff
@@ -280,28 +249,17 @@ def remove_contracts(label: str):
                     raise ValueError(f"End of range subset, {labelSpecifier.end}, not found in pipeline "
                                      "definition")
 
-            closed = False
+            labelSet = set()
             for label in labels:
-                # if there is a begin label delete all labels until it is
-                # reached.
                 if labelSpecifier.begin:
                     if label != labelSpecifier.begin:
-                        pipeline.removeTask(label)
-                        remove_contracts(label)
                         continue
                     else:
                         labelSpecifier.begin = None
-                # if there is an end specifier, keep all tasks until the
-                # end specifier is reached, afterwards delete the labels
-                if labelSpecifier.end:
-                    if label != labelSpecifier.end:
-                        if closed:
-                            pipeline.removeTask(label)
-                            remove_contracts(label)
-                        continue
-                    else:
-                        closed = True
-        return pipeline
+                labelSet.add(label)
+                if labelSpecifier.end and label == labelSpecifier.end:
```

Comment:
If the first test here could be `labelSpecifier.end is not None`, it probably should be.